crates/nu-protocol/src/deprecation.rs

```diff
@@ -87,7 +87,16 @@ impl DeprecationEntry {
     fn check(&self, call: &Call) -> bool {
         match &self.ty {
             DeprecationType::Command => true,
-            DeprecationType::Flag(flag) => call.get_named_arg(flag).is_some(),
+            DeprecationType::Flag(flag) => {
+                // Make sure we don't accidentally have dashes in the flag
+                debug_assert!(
+                    !flag.starts_with("--"),
```

Comment:
Being defensive ;)

```suggestion
                    !flag.starts_with("-"),
```

Reply:
oh shoot thanks, i originally was going to have a different message for `--` and `-` but changed my mind